src/main/java/io/cdap/plugin/SFTPPutAction.java

```diff
@@ -85,33 +86,41 @@ public void run(ActionContext context) throws Exception {
     if (!fileSystem.exists(source)) {
       throw new RuntimeException(String.format("Source Path doesn't exist at %s", source));
     }
-
-    try (SFTPConnector sftp = new SFTPConnector(config.getHost(), config.getPort(), config.getUserName(),
-                                                config.getPassword(), config.getSSHProperties())) {
-      ChannelSftp channel = sftp.getSftpChannel();
-
-      try {
-        channel.mkdir(config.getDestDirectory());
-      } catch (SftpException ex) {
-        // Suppress since the directory might already exist.
+    if (config.getAuthTypeBeingUsed().equals("privatekey-select")) {
+      try (SFTPConnector sftp = new SFTPConnector(config.getHost(), config.getPort(), config.getUserName(),
+        config.getPrivateKey(), config.getPassphrase(), config.getSSHProperties())) {
+        sftpPutLogic(fileSystem, source, sftp);
+      } catch (Exception e) {
+        LOG.error(String.valueOf(e));
       }
+    } else {
+      try (SFTPConnector sftp = new SFTPConnector(config.getHost(), config.getPort(), config.getUserName(),
+        config.getPassword(), config.getSSHProperties())) {
+        sftpPutLogic(fileSystem, source, sftp);
+      } catch (Exception e) {
+        LOG.error(String.valueOf(e));
+      }
+    }
+  }
 
-      channel.cd(config.getDestDirectory());
-
-      // Filter out only the files to copy
-      FileStatus[] filesToCopy = fileSystem.listStatus(source, new PathFilter() {
-        @Override
-        public boolean accept(Path path) {
-          String fileName = path.getName();
-          return fileName.matches(config.getFileNameRegex());
-        }
-      });
-
-      for (FileStatus file : filesToCopy) {
-        Path filePath = file.getPath();
-        try (InputStream inputStream = fileSystem.open(filePath)) {
-          channel.put(inputStream, filePath.getName());
-        }
+  private void sftpPutLogic(FileSystem fileSystem, Path source, SFTPConnector sftp)
+    throws SftpException, IOException {
+    ChannelSftp channel = sftp.getSftpChannel();
+    try {
+      channel.mkdir(config.getDestDirectory());
+    } catch (SftpException ex) {
```

Comment:
we should catch specific exception if dir already exists or rely on channel to find it. This exception may indicate underlying issue and should not be suppressed

Reply:
fixed

Reply:
what is the fix here?

Reply:
if the directory already exists it outputs a warning though this doesnt effect the running of the action. This warning was happening every test run for me since I was not having the action create a directory

Reply:
Look at `SftpException` class. Looks like there is no flag to detect if the dir already exists

May be we can do something like following:
https://stackoverflow.com/a/20121303/11408628